version/version.go

```diff
@@ -17,19 +19,48 @@ import (
 
 //go:embed VERSION
 var baseVersion string
-var buildVersion string
+var buildNumber string
 
 func Version() string {
 	return strings.TrimSpace(baseVersion)
 }
 
-func BuildVersion() string {
-	if buildVersion == "" {
+func BuildNumber() string {
+	if buildNumber == "" {
 		return "x"
 	}
-	return buildVersion
+	return buildNumber
+}
+
+func commitInfo() string {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "x"
+	}
+
+	dirty := ".dirty"
+	var commit string
+	for _, setting := range info.Settings {
+		switch setting.Key {
+		case "vcs.revision":
+			commit = setting.Value
+		case "vcs.modified":
+			if setting.Value == "false" {
+				dirty = ""
+			}
+		}
```

Comment:
I tend to prefer whatever has fewer indents. Here, that was the if/else if chain.

But it's a not opinion I hold very strongly.